### ISSUE TITLE
HTTPS binding failes to set credential

### ISSUE THREAD

**Issue by Liran-Dobrish** [Microsoft/vsts-rm-extensions#57]
hello
I'm using the task to deploy my web site to IIS 8 on Windows2012R2.
i first deploy the site with http and application pool configurations set.
i then run the task again to set the Https binding only.
and it fails with "incorrect parameter" error.
the log show the following:
"_2016-07-18T13:27:18.0674249Z Checking website exists. Running command : "C:\Windows\system32\inetsrv\appcmd.exe"  list site /name:"MySite.com"
2016-07-18T13:27:18.0674249Z Website ("MySite.com") already exists
2016-07-18T13:27:18.0674249Z Checking website physical path exists c:\Repositories\MySite\1.0.1.182\MySite
2016-07-18T13:27:18.0674249Z Checking binding exists for website ("MySite.com"). Running command : "C:\Windows\system32\inetsrv\appcmd.exe"  list sites
2016-07-18T13:27:18.0684248Z Does bindings exist for website ("MySite.com") is : False
2016-07-18T13:27:18.0684248Z Updating website properties. Running command : "C:\Windows\system32\inetsrv\appcmd.exe"  set site /site.name:"MySite.com" -[path='/'].[path='/'].physicalPath:"c:\Repositories\MySite\1.0.1.182\MySite" /+bindings.[protocol='https',bindingInformation='*:443:MySite.com']
2016-07-18T13:27:18.0684248Z Checking if SslCert binding is already present. Running command : netsh http show sslcert ipport=0.0.0.0:443
2016-07-18T13:27:18.0684248Z Setting SslCert for website.
2016-07-18T13:27:18.0734246Z Deployment failed on machine spiderman6 with following message : System.Exception: The parameter is incorrect.  
2016-07-18T13:27:18.0834269Z ##[error]Microsoft.PowerShell.Commands.WriteErrorException: Deployment on one or more machines failed.
2016-07-18T13:27:18.0844269Z ##[error]For more info please refer to http://aka.ms/iisextnreadme_"

i ran the same commands on the machine and i got error 183 that the certification already exists
as the computer contains several other certifications and the first one is selected by default when setting https binding

I'm using the latest version found in the TFS Marketplace (1.2.1).

**Comment by mvvsubbu** [comment]
Hi Liran,

Sorry for the delay in response. Can we have a call regarding this; for me 9 AM - 6 PM IST works. You can mail me at subraman at Microsoft dot com with your suitable time.

**Comment by mvvsubbu** [comment]
Did you get unblocked from this issue?

**Comment by Christoffleroux** [comment]
Im having the same issue setting the certificate on the created binding. I have attached my task log.
[tasklog_7.zip](https://github.com/Microsoft/vsts-rm-extensions/files/404955/tasklog_7.zip)

**Comment by niadak** [comment]
Can you please try the following command in the target machine.

netsh http add sslcert hostnameport=ww.tstcore.smd.co.za:5555 certhash=2A756F641B02C1971FFA1D34FEDE43D7C6FFD9BC appid={{497D490F-EEA7-4F2B-AB94-48D9C1ACDCB1}} certstorename=MY

If that succeeded, please varify the certificate Thumb print you are providing in task input. From the logs it seems like there is a extra ? in the Thumb Print.

**Comment by Christoffleroux** [comment]
Hi Niadak

I’ve run the command on our target server (Win Server 2012 R2)

Receiving the following

[cid:image001.png@01D1F47F.40292FC0]

Thanks for the support
Christoff

**Comment by niadak** [comment]
@Christoffleroux , 
I am assuming you are unblocked and closing the issue. Please feel free to reopen otherwise

**Comment by Christoffleroux** [comment]
Hi Niadak, no please dont close this issue, im still having this problem, ive sent you the output from the command you requested.

**Comment by mvvsubbu** [comment]
Can we have a call to close the issue faster, Any time between 9 AM to 6 PM IST is fine, you can reply with your convenient time so that we can have a call

**Comment by mvvsubbu** [comment]
Can you respond with your time slot for meeting

**Comment by mvvsubbu** [comment]
I am closing the issue, if you have still have the issue contact RM_Customer_Queries at Microsoft dot com

**Comment by Sacha-K** [comment]
RM_Customer_Queries at Microsoft dot com is redirected to vsdpteng at microsoft dot com that is only accepting Microsoft internal emails and throwing a 550.

**Comment by Sacha-K** [comment]
I’m using the WinRM - IIS Web App Management release task with a TFS on premise server. I’m trying to add a https binding to a website but there’s something wrong with the SSL Certificate Thumb Print. I set it as ‎24b0ade3c0f8d16abaae0f31bc5ad161f9692cd9 but the release fails and I only have the following in the logs:

> 2016-10-07T08:35:47.2883787Z Executing the powershell script: C:\Agent1\tasks\IISWebAppMgmt\1.0.6\Main.ps1
> 2016-10-07T08:35:47.5696275Z Started creating website: Default Web Site
> 2016-10-07T08:35:47.5852591Z Performing deployment in parallel on all the machines.
> 2016-10-07T08:35:47.5852591Z Deployment started for machine: XYZ with port 5986.
> 2016-10-07T08:35:58.1638798Z Deployment status for machine XYZ : Microsoft.VisualStudio.Services.DevTestLabs.Definition.DeploymentResponse.Status
> **2016-10-07T08:35:58.1638798Z Deployment failed on machine XYZ with following message : System.Exception: The parameter is incorrect.**
> 2016-10-07T08:35:58.1795067Z ##[error]Deployment on one or more machines failed.
> 2016-10-07T08:35:58.1795067Z ##[error]For more info please refer to http://aka.ms/iisextnreadme

What is the correct format for the certificate thumb print?

**Comment by Sacha-K** [comment]
Looks like it was caused by an extra ? at the beginning of the thumbprint for everyone here. I copied the thumbprint from the Details tab in the certificate view on the server and somehow ended up with the extra ? at the beginning.

It's some invisible unicode character. It's a bit sneaky as it looks fine in TFS, then I modified AppCmdOnTargetMachines.ps1 to add a Write-Verbose showing the command and it also looks fine, the ? only becomes visible when you copy the thumbprint from the release configuration or the error log in a text editor setup as ANSI.
